.changelog/2640.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+resource/mongodbatlas_advanced_cluster: Enforces `priority` descending order in `region_configs`
```

Comment:
```suggestion
resource/mongodbatlas_advanced_cluster: Enforces `priority` descending order in `region_configs` avoiding potential non-empty plans after apply
```
we can add more detail of the why

Reply:
done in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2640/commits/bfb2e969d42caa25b8cabccfc2e21d0dddf71ec5